chain/types/actor_event.go

```diff
@@ -65,3 +81,286 @@ type ActorEvent struct {
 	// CID of message that produced this event.
 	MsgCid cid.Cid `json:"msgCid"`
 }
+
+// AsCompactEncoded will trigger alternate JSON encoding for ActorEvents, where the event entries
+// are encoded as a list of tuple representation structs, rather than a list of maps, values are
+// decoded using the specified codec where possible, and they are encoded using dag-json form so
+// bytes are represented using the `{"/":{"bytes":"base64"}}` form rather than Go standard base64
+// encoding.
+func (ae ActorEvent) AsCompactEncoded() ActorEvent {
```

Comment:
Why do we need this ? I don't see us not using this in any of the APIs. Is it so that we can ultimately allow users to specify this ? I think the best thing to do would be to ship this PR with 1.26 so we can get rid of this optionality.

Reply:
Yeah, I hate the switching here, agree with @aarshkshah1992